packages/adapter-next/src/hooks/customType-library-read.ts

```diff
@@ -0,0 +1,46 @@
+import type { CustomTypeLibraryReadHook } from "@slicemachine/plugin-kit";
+import * as prismicT from "@prismicio/types";
+import * as fs from "node:fs/promises";
+import * as path from "path";
+
+import { readJSONFile } from "../lib/readJSONFile";
+
+import type { PluginOptions } from "../types";
+
+const isCustomTypeModel = (
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	input: any,
+): input is prismicT.CustomTypeModel => {
+	return typeof input === "object" && input !== null && "json" in input;
+};
+
+export const customTypeLibraryRead: CustomTypeLibraryReadHook<
+	PluginOptions
+> = async (_data, { helpers }) => {
+	const dirPath = helpers.joinPathFromRoot("customtypes");
+
+	const childFiles = await fs.readdir(dirPath);
+
+	const ids: string[] = [];
+	await Promise.all(
+		childFiles
```

Comment:
In current SM projects, `customtypes` uses the following structure:

`/customtypes/${apiID}/index.json`

So we'll need to change some of the logic here.

I don't think we want to make a breaking change here since that would require existing projects to relocate their Custom Type JSON model files.

Reply:
Right, got confused by the read one ☺️